frontend/include/chpl/framework/ErrorBase.h

```diff
@@ -176,20 +188,21 @@ class BasicError : public ErrorBase {
 };
 
 /**
-  An error originating from the parser.
+   A parsing error simple enough that no error class is specialized for it.
  */
 class ParseError : public BasicError {
  protected:
-  ParseError(Kind kind, ID id, Location loc, std::string message, std::vector<Note> notes)
-    : BasicError(kind, PARSE, std::move(id), std::move(loc), std::move(message),
-        std::move(notes)) {}
-
-  static const owned<ParseError>&
-  getParseError(Context* context, Kind kind, ID id,
-                Location loc, std::string message,
-                std::vector<Note> notes);
+  ParseError(const Location loc, const std::string message)
```

Comment:
Hi, I'm just peeking in. Why put `const` on immediately-moved by-value constructor parameters? I'm no expert on C++ semantics, but I'm reasonably sure that `const` things can't be `move`'d, which means this change will require an extra copy to pass arguments to `BasicError`.

Reply:
I think that's right, I'll change it. [sort-of reference](https://www.nextptr.com/tutorial/ta1211389378/beware-of-using-stdmove-on-a-const-lvalue)